CI: Add post-merge stage to deploy Storybook

carlito-scality opened the pull request:
**Component**:CI

<!-- E.g. 'Layout', 'Table', 'build', 'tests'... -->

**Description**:
- Add post-merge stage to deploy Storybook in the CI instead of using Github Actions

**Design**:

**Breaking Changes**:

- [] Breaking Changes


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #ISSUE_NUMBER

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

carlito-scality committed:
Remove Github Actions workflows due to organisation plan

bert-e commented:
# Hello carlito-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/core-ui/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @carlito-scality

* @chengyanjin

* @rachedbenmustapha

* @nicolas2bert

* @monpote

tcarmet commented on eve/main.yml:
```diff
@@ -39,3 +41,34 @@ stages:
             npm run flow &&
             npm run lint
           haltOnFailure: false
+  post-merge:
```
For now, there's no need to split your post-merge stage into 2, I would directly put the `storybook_deploy` stage like the following:
```
 development/*:
    stage: storybook_deploy
```

or rename the `storybook_deploy` as `post-merge`

tcarmet commented on eve/main.yml:
```diff
@@ -39,3 +41,34 @@ stages:
+          stage_names:
+            - storybook_deploy
+          haltOnFailure: true
+  storybook_deploy:
```
nit: I find it weird to use `-` and `_` as a seperator to name things, either use `-` everywhere or `_`

but this is just a personal thought, feel free to do whatever.

tcarmet commented on eve/main.yml:
```diff
@@ -39,3 +41,34 @@ stages:
+          context: '.'
+          dockerfile: eve/workers/pod-unit-tests/Dockerfile
+    steps:
+      - Git: &git_pull
```
you can replace this whole step as
`- Git: *git_pull` and that's it, remove all the other lines.

here you're creating two reference with the `&` character, which makes your `yaml` wrong

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @carlito-scality

* @chengyanjin

* @rachedbenmustapha

* @nicolas2bert

* @monpote





The following reviewers are expecting changes from the author, or must review again:

* @tcarmet

tcarmet commented on eve/main.yml:
```diff
@@ -39,3 +41,34 @@ stages:
+          command: >
+            npm ci &&
+            npm run storybook:deploy -- --ci --host-token-env-variable=%(secret:core_ui_github_token)s
+          haltOnFailure: false
```
could you add here the following parameter
```
env:
  GIT_CONFIG_NOSYSTEM: "1"
```

carlito-scality committed:
CI: Add post-merge stage to deploy Storybook
- Use internal CI instead of Github Actions
- Move GITHUB_TOKEN to envrionment variables
- Add simultaneous_builds
- ensure that one post-merge will be executed at a time.
- Add worker_unit_tests  worker reference

tcarmet commented on eve/main.yml:
```diff
@@ -39,3 +32,22 @@ stages:
+      images:
+        docker-unit-tests:
+          context: '.'
+          dockerfile: eve/workers/pod-unit-tests/Dockerfile
```
Could you explain to me why the context needs to be `.` because I took a look at the dockerfile, and it doesn't seem to be needed. You could even replace those lines to :
```yaml
docker-unit-tests: 'eve/workers/pod-unit-tests'
```

and same story for the one on the `pre-merge` as it's the same worker.

tcarmet commented on eve/main.yml:
```diff
@@ -39,3 +30,20 @@ stages:
             npm run lint
           haltOnFailure: false
+  post-merge:
+    worker:
```
could you add, at the same lever as the worker the option `simultaneous_builds`. Like the following:

```yaml
post-merge:
  simultaneous_builds: 1
  worker:
...
```
This will ensure that only one post-merge will be executed at a time. Per example if 5 merge happen, one after the other, they will not be running over each other, with this you ensure that the content of your page will be coherent.

tcarmet commented on eve/main.yml:
```diff
@@ -4,25 +4,16 @@ version: "0.2"
-            - unit_tests
-          haltOnFailure: true
-  unit_tests:
     worker:
```
could you add a reference, here to the worker.
Like the following `worker: &worker_unit_tests` (you can name it however you want).

once you added it, modify the content of the worker on the `post-merge` stage with the following:
```yaml
post-merge:
  worker: *worker_unit_tests
```

tcarmet commented on eve/main.yml:
```diff
@@ -39,3 +30,22 @@ stages:
+            npm ci &&
+            npm run storybook:deploy -- --ci --host-token-env-variable=GITHUB_TOKEN
+          env:
+            GIT_CONFIG_NOSYSTEM: "1"
```
Could you just add the following comment above the `GIT_CONFIG_NOSYSTEM`

```
# Our system gitconfig is setup to go through a gitcache service and
# as we're preforming write operations with git on this step, we don't want that the request
# go through the gitcache.
```

tcarmet replied:
it's a reminder in case someone wants to remove this option in the future :P

tcarmet reviewed (approved):
one last request to add a comment. Otherwise it's looking good! I'm gonna run a little test and come back with results

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @carlito-scality

* @chengyanjin

* @rachedbenmustapha

* @nicolas2bert

* @monpote

carlito-scality commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.0`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`




Please check the status of the associated issue None.

Goodbye carlito-scality.
